csrc/iter_visitor.cpp

```diff
@@ -1102,9 +1107,9 @@ void DeadCodeRemover::markLiveRecursive(Statement* stmt) {
     return;
   }
   markLive(stmt);
-  if (stmt->isVal() && stmt->asVal()->definition()) {
-    markLiveRecursive(stmt);
-  } else {
+  if (stmt->isVal() && stmt->asVal()->definition() != nullptr) {
+    markLiveRecursive(stmt->asVal()->definition());
```

Comment:
is this just trying to avoid marking stmt live again, since we already called `markLive(stmt)`?

Reply:
This was a bug where we were actually not recursing into the definition actually!